rust-toolchain.toml

```diff
@@ -0,0 +1,16 @@
+[toolchain]
+# Pinned due to https://github.com/rust-osdev/uefi-rs/issues/500.
+#
+# Compilation started failing 2022-09-02. The previous several builds
+# don't have miri available though, so pin to a slightly older version.
+channel = "nightly-2022-08-26"
+components = [
+    # Needed for `cargo xtask miri`.
+    "miri",
+    # Needed for `-Zbuild-std`.
+    "rust-src",
+
+    # Standard components.
+    "clippy",
+    "rustfmt",
+]
```

Comment:
Filed an issue for removing build-std stuff: https://github.com/rust-osdev/uefi-rs/issues/504

I don't think adding `targets = [ "x86_64-unknown-uefi" ]` would be the right thing to do for two reasons:
1. There are multiple UEFI targets -- uefi-rs works for `aarch64-unknown-uefi` and `i686-unknown-uefi` as well
2. The `xtask` package runs on the host, so it needs to build with the host target.

Reply:
The targets-field is confusing as it is not properly documented. From my experience I can say: We actually want to add ALL uefi targets there (if they are Tier 2 or above eventually). Everything this "targets" property says is:

"If someone executes `cargo build --target foo`, execute `rustup target add foo` if it is not yet installed. Otherwise, just continue."

Thus, it is totally fine to list targets there, even if you never use it via `cargo build --target foo`. This should not be confused with 

.cargo/config.toml
```toml
[build]
target = "foo"
```

which sets `--target foo` as default for all cargo commands.

I just had this discussion with colleagues a few weeks ago, when I introduced a rust-toolchain.toml into our project :)

Reply:
:+1: Gotcha, I did confuse the two config files there. I've added a comment to #504 noting that we should add this field to `rust-toolchain.toml`.